### ISSUE TITLE
Global declaration of `Observable` causes issues when using latest rxjs 

### ISSUE THREAD

**Issue by ntziolis** [apollographql/apollo-client#7839]
The globally declared `Observable` interface is not compatible with `rxjs/Observable`. This causes the following error when declaring variables of type `Observable`:  
> error "Property '['@@observable']' is missing in type rxjs/Observable

While this can be resolved by manually adding the import. It breaks editor auto import functionality as the editor thinks `Observable` is already a known type. This is a major pain as `rxjs/Observable` is used in nearly everywhere (angular app).

We are currently using patch-package `@apollo/client@3.3.11` to completely remove the global declaration. This re-enables normal editor flow without us noticing any side effects.

While this is an [issue](https://github.com/kamilkisiela/apollo-angular/issues/1579) that is angular related its not something that can be solved in the `apollo-angular` package alone. Hence filing this issue here.

We currently assume that the current versions of `rxjs/Observable` is simply not compatible with the global declaration anymore.

Lib versions:
```
    "@apollo/client": "^3.3.11",
    "apollo-angular": "^2.4.0",
    "rxjs": "~6.6.3",
```
Here is the diff from patch-package:
```diff
diff --git a/node_modules/@apollo/client/utilities/observables/Observable.d.ts b/node_modules/@apollo/client/utilities/observables/Observable.d.ts
index dd1ed9a..56f471f 100644
--- a/node_modules/@apollo/client/utilities/observables/Observable.d.ts
+++ b/node_modules/@apollo/client/utilities/observables/Observable.d.ts
@@ -3,10 +3,10 @@ import 'symbol-observable';
 export declare type ObservableSubscription = ZenObservable.Subscription;
 export declare type Observer<T> = ZenObservable.Observer<T>;
 export declare type Subscriber<T> = ZenObservable.Subscriber<T>;
-declare global {
-    interface Observable<T> {
-        ['@@observable'](): Observable<T>;
-    }
-}
+// declare global {
+//     interface Observable<T> {
+//         //['@@observable'](): Observable<T>;
+//     }
+// }
 export { Observable };
 //# sourceMappingURL=Observable.d.ts.map
\ No newline at end of file
```

**Comment by benjamn** [comment]
@ntziolis Thanks for bringing this to our attention. Fix incoming: #7888

**Comment by ntziolis** [comment]
That was wicked fast!
